src/ui/mod.rs

```diff
@@ -75,7 +76,36 @@ impl<R: ResourcesProvider> Ui<R> {
 }
 
 pub fn print_error(err: &Error) {
-    println!("{:?}", err);
+    for (i, error) in err.chain().enumerate() {
+        let prefix = if i == 0 {
+            "".to_owned()
+        } else {
+            "\t".to_owned()
+        };
+        println!("{}", add_indentation(&error.to_string(), &prefix));
+        if let Some(hint) = get_error_hint(error) {
+            let prefix = prefix.to_owned() + "\t";
+            println!("{}\n", add_indentation("Hint:", &prefix));
+            println!("{}", add_indentation(&hint, &prefix));
+        }
+    }
+}
+
+fn get_error_hint(error: &(dyn std::error::Error + 'static)) -> Option<&'static str> {
+    match error.downcast_ref::<ApiError>() {
+        Some(ApiError::LanguageDeductionError(_)) => Some(LANGUAGE_HINT),
+        Some(ApiError::TaskDeductionError(_)) => Some(TASK_HINT),
+        _ => None,
+    }
+}
+
+fn add_indentation(text: &str, prefix: &str) -> String {
+    let mut result = String::new();
+    for line in text.split_inclusive('\n') {
+        result.push_str(prefix);
+        result.push_str(&line);
```

Comment:
```suggestion
        result.push_str(line);
```
This should be a clippy lint :)